src/compiler/checker.ts

```diff
@@ -2958,6 +2962,9 @@ namespace ts {
             }
 
             function typeToTypeNodeHelper(type: Type, context: NodeBuilderContext): TypeNode {
+                if (cancellationToken && cancellationToken.throwIfCancellationRequested) {

```

Comment:
checkout ```runWithCancellationToken``` in program.ts, it throws away checker.. do we need to do that . I assume not since we arent caching anything.. I think that should be then reflected in the api. 
Also add a test case that it actually cancels. (eg. there is test in harness/unittests/builder.ts that checks cancellation token works.